common/src/main/java/com/wynntils/models/beacons/BeaconModel.java

```diff
@@ -72,5 +107,62 @@ public void onEntityRemoved(RemoveEntitiesEvent event) {
             beacons.remove(entityId);
             WynntilsMod.postEvent(new BeaconEvent.Removed(removedBeacon));
         });
+
+        event.getEntityIds().stream().filter(beaconMarkers::containsKey).forEach(entityId -> {
+            BeaconMarker removedBeaconMarker = beaconMarkers.get(entityId);
+            beaconMarkers.remove(entityId);
+            WynntilsMod.postEvent(new BeaconMarkerEvent.Removed(removedBeaconMarker));
+        });
+    }
+
+    public void registerBeacon(String name, BeaconKind beaconKind) {
+        beaconRegistry.put(name, beaconKind);
+    }
+
+    public void registerBeaconMarker(String name, BeaconMarkerKind beaconMarkerKind) {
+        beaconMarkerRegistry.put(name, beaconMarkerKind);
+    }
```

Comment:
I think the interfaces could have `name` methods on it, and when implementing on an enum, it'll work automatically.